### ISSUE TITLE
Why I follow the Linear Scaling Rule to adjust my Lr, but my loss gets nan?

### ISSUE THREAD

**Issue by minixiguazi** [open-mmlab/mmdetection#1710]
There is a statement in GETTING_STARTED.md as following:

***Important**: The default learning rate in config files is for 8 GPUs and 2 img/gpu (batch size = 82 = 16). According to the Linear Scaling Rule, you need to set the learning rate proportional to the batch size if you use different GPUs or images per GPU, e.g., lr=0.01 for 4 GPUs * 2 img/gpu and lr=0.08 for 16 GPUs * 4 img/gpu.

But when I train the source code, I set lr=0.12 for 48 GPUs * 2 img/gpu and my loss get nan. I reduce my lr to 0.08 and 0.06, my loss still get nan. lr = 0.04 will be better.
Why should this situation happen? Can you tell me why please? 
Thank you very much!

**Comment by ZwwWayne** [comment]
This is because at this time the LR is too big.
You can make the warm-up iters=1000 or more, and use `exp` rather than ` linear` for warm-up to make the warm-up process more stable.

**Comment by minixiguazi** [comment]
Can you give me an example?  
 
Config files is for 8 GPUs and 2 img/gpu:

optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=1.0 / 3,
    step=[8, 11])

**Comment by ZwwWayne** [comment]
warmup='exp',
warmup_iters=1000,
warmup_ratio=1.0/10,
Usually, these three modifications are enough, or you could make the `warmup_iters=2000` or greater.

**Comment by minixiguazi** [comment]
optimizer = dict(type='SGD', lr=0.12, momentum=0.9, weight_decay=0.0001)
where lr=0.02 changes to 0.12?

**Comment by ZwwWayne** [comment]
Yes, if you need to scale the batch size x6.

**Comment by minixiguazi** [comment]
thanks you

**Comment by ZwwWayne** [comment]
If it blows up, you can also change the max_norm=35 to max_norm=10.
That is all I can help, the remaining thing is tuning parameters.
So I will close this issue, feel free to reopen it if you have any further questions.
